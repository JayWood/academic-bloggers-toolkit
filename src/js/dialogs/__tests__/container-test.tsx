import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { observable } from 'mobx';
import * as React from 'react';
import Container from '../container';

const setup = () => {
    const currentDialog = observable('hello');
    const component = shallow(
        <Container title="Hello World" currentDialog={currentDialog}>
            <h1>Hello World</h1>
        </Container>,
    );
    return {
        component,
        instance: component.instance() as Container,
        currentDialog,
    };
};

describe('<Container />', () => {
    it('should match snapshot', () => {
        const { component } = setup();
        expect(toJSON(component)).toMatchSnapshot();
    });
    it('should render', () => {
        const { component } = setup();
        expect(component).toBeTruthy();
    });
    it('should handle mouse wheel appropriately', () => {
        const { component } = setup();
        const preventDefault = jest.fn();
        component.simulate('wheel', { cancelable: false, preventDefault });
        expect(preventDefault).not.toHaveBeenCalled();
        component.simulate('wheel', { cancelable: true, preventDefault });
        expect(preventDefault).toHaveBeenCalledTimes(1);
    });
    it('should close when escape is pressed', () => {
        const { component, currentDialog } = setup();
        const stopPropagation = jest.fn();
        component.simulate('keyDown', { key: 'a', stopPropagation });
        expect(currentDialog.get()).toBe('hello');
        expect(stopPropagation).not.toHaveBeenCalled();
        component.simulate('keyDown', { key: 'Escape', stopPropagation });
        expect(currentDialog.get()).toBe('');
        expect(stopPropagation).toHaveBeenCalledTimes(1);
    });
    it('should close with and without an associated event', () => {
        const { instance } = setup();
        expect(instance.close).not.toThrow();
        const preventDefault = jest.fn();
        instance.close({ preventDefault } as any);
        expect(preventDefault).toHaveBeenCalled();
    });
});
