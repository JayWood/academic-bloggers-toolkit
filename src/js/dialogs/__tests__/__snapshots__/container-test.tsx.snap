// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should match snapshot 1`] = `
<div
  className="dialog"
  onKeyDown={[Function]}
  onWheel={[Function]}
  style={
    Object {
      "background": "rgba(0, 0, 0, 0.7)",
    }
  }
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={Object {}}
    paused={false}
    tag="div"
  >
    <div
      aria-labelledby="dialog-label-Hello_World"
      className="dialog__main"
      role="dialog"
      style={
        Object {
          "width": 600,
        }
      }
    >
      <header>
        <span
          className="dialog__title"
          id="dialog-label-Hello_World"
        >
          Hello World
        </span>
        <button
          aria-label="Close dialog"
          onClick={[Function]}
        >
          <span
            className="dashicons dashicons-no-alt"
          />
        </button>
      </header>
      <div
        className="dialog__body"
      >
        <h1
          focusTrapPaused={false}
        >
          Hello World
        </h1>
      </div>
    </div>
  </FocusTrap>
  <style
    jsx={true}
  >
    
                    header {
                        height: 40px;
                        font-size: 16px;
                        font-weight: 500;
                        color: #23282d;
                        display: flex;
                        justify-content: space-between;
                        line-height: 40px;
                        background: #f5f5f5;
                        border-radius: 4px 4px 0 0;
                        box-shadow: 0 1px 0 rgba(16, 22, 26, 0.15);
                    }
                    button {
                        cursor: pointer;
                        border: 0;
                        outline: 0;
                        padding: 0;
                        background: transparent;
                        color: #555d66;
                        width: 40px;
                    }
                    button:hover {
                        color: black;
                    }
                    button:focus {
                        outline: 1.1px auto rgba(19, 124, 189, 0.5);
                    }
                    .dialog {
                        position: fixed;
                        height: 100vh;
                        width: 100vw;
                        top: 0;
                        left: 0;
                        z-index: 9999;
                        display: flex;
                        flex-direction: column;
                        justify-content: center;
                        align-items: center;
                    }
                    .dialog__main {
                        height: auto;
                        margin-bottom: -32px;
                        border-radius: 4px;
                        background: white;
                        box-shadow: 0 10px 20px rgba(0, 0, 0, .19), 0 6px 6px rgba(0, 0, 0, .23);
                    }
                    .dialog__title {
                        margin: 0 10px;
                    }
                    .dashicons-no-alt {
                        font-size: 25px;
                        width: 25px;
                        height: 25px;
                    }
                
  </style>
</div>
`;
